Detectors/PHOS/reconstruction/src/RawReaderMemory.cxx

```diff
@@ -34,15 +34,11 @@ void RawReaderMemory::setRawMemory(const gsl::span<const char> rawmemory)
 o2::header::RDHAny RawReaderMemory::decodeRawHeader(const void* payloadwords)
 {
   auto headerversion = RDHDecoder::getVersion(payloadwords);
-  if (headerversion == 4) {
-    return o2::header::RDHAny(*reinterpret_cast<const o2::header::RAWDataHeaderV4*>(payloadwords));
-  } else if (headerversion == 5) {
-    return o2::header::RDHAny(*reinterpret_cast<const o2::header::RAWDataHeaderV5*>(payloadwords));
-  } else if (headerversion == 6) {
-    return o2::header::RDHAny(*reinterpret_cast<const o2::header::RAWDataHeaderV6*>(payloadwords));
+  if (headerversion < RDHDecoder::getVersion<o2::header::RDHLowest>() || headerversion > RDHDecoder::getVersion<o2::header::RDHHighest>()) {
+    LOG(error) << "Wrong header version " << headerversion;
+    throw RawDecodingError::ErrorType_t::HEADER_DECODING;
   }
-  LOG(error) << "Wrong header version " << int(headerversion);
-  throw RawDecodingError::ErrorType_t::HEADER_DECODING;
```

Comment:
Should this be removed??? I think this is copy-past error?

Reply:
Well, I've just preserved your error message and exception I case of a wrong header version. Do you want to change this behaviour?

Reply:
I think Log and through should be within if starting in L37 and should not appear in normal flow

Reply:
But they are within the `if` checking for the wrong version

Reply:
@peressounko do you still have some doubts or I should merge it?